lib/pi_piper/pin.rb

```diff
@@ -154,12 +157,23 @@ def wait_for_change
     # In short, you must call this method if you are curious about the 
     # current state of the pin.
     def read
+      fail PiPiper::PinError, "Pin #{@pin} already released" if released?
       @last_value = @value
       val = Platform.driver.pin_read(@pin)
       @value = invert ? (val ^ 1) : val
     end
 
+    def release
+      Platform.driver.release_pin(@pin)
+      @released = true
+    end
+
+    def released?
+      @released == true
```

Comment:
I see, something like `Platform.driver.released?(pin)`. I can change it that way.